ecommerce_extensions/payment/processors/edupay.py

```diff
@@ -0,0 +1,384 @@
+"""eduNEXT custom payment processor file.
+
+Contain a configurable payment processor for multiples custom processors, the
+configuration will set the transaction parameters and custom requirements for
+every different processors, ONLY one configuration can be set by site so it is
+not possible multiple custom processors by site.
+
+Classes:
+    EdnxPaymentProcessor: Configurable payment class processor.
+"""
+import logging
+from decimal import Decimal
+from hashlib import sha256
+
+import waffle
+from crum import get_current_request
+from django.conf import settings
+from django.urls import reverse
+from django.utils.decorators import classproperty
+from django.utils.functional import cached_property
+from oscar.apps.payment.exceptions import GatewayError, TransactionDeclined
+from oscar.core.loading import get_model
+
+from ecommerce_extensions.payment.constants import InvalidEdnxDecision, TransactionPending
+from ecommerce_extensions.payment.encryption_utils import encode_string
+
+try:
+    from ecommerce.extensions.payment.exceptions import InvalidSignatureError
+    from ecommerce.extensions.payment.processors import BasePaymentProcessor, HandledProcessorResponse
+except ImportError:
+    BasePaymentProcessor = object
+
+logger = logging.getLogger(__name__)
+
+PaymentEvent = get_model('order', 'PaymentEvent')
+PaymentEventType = get_model('order', 'PaymentEventType')
+ProductClass = get_model('catalogue', 'ProductClass')
+Source = get_model('payment', 'Source')
+SourceType = get_model('payment', 'SourceType')
+DEFAULT_NAME = 'edupay'
+
+
+class EdnxPP(BasePaymentProcessor):
```

Comment:
Please use more descriptive names

Reply:
EdnxPaymentProcessor? what do you think?